code/modules/mob/mob_helpers.dm

```diff
@@ -131,6 +131,41 @@ var/global/list/limb_types_by_name = list(
 		index++
 	return output_message
 
+// proc that parses an html input string and scrambles the non-html string contents
+/proc/stars_decode_html(message)
+	if(!length(message))
+		return
+
+	// todo: sanitize string more to remove unnencessary <>.
+	// list of parsed strings that have been scrambled, used to then reinsert the scrambled strings in to the message
+	var/list/replaced_scrambled_strings = list()
+
+	// boolean value to know if the current indexed element needs to be added to the scrambled message.
+	var/parsing_message = FALSE
+
+	// the current string value that needs to be scrambled.
+	var/current_string_to_scramble = ""
+
+	for(var/character_index in 1 to length(message))
+		if(message[character_index] == ">")
+			parsing_message = TRUE
+			continue
+		if(message[character_index] == "<")
+			parsing_message = FALSE
```

Comment:
You also need to check for quotation marks inside of HTML tags. Consider a tag such as `<tag property=">Injection()" />`. So when you are inside an HTML tag and run into a quotation mark (either single `'` or double `"`), then you advance until you find the matching quotation mark. Perhaps you could add a couple booleans for this, e.g. `need_single_quote` and `need_double_quote`.

Maybe it would be a bit overkill but it's probably good to be safe.

Reply:
Alright.